src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -566,9 +566,12 @@ impl Visitor<'tcx> for Validator<'_, 'mir, 'tcx> {
             StatementKind::FakeRead(FakeReadCause::ForMatchedPlace, _) => {
                 self.check_op(ops::IfOrMatch);
             }
+            StatementKind::SetDiscriminant { ref place, .. } => {
+                let ctx = PlaceContext::MutatingUse(MutatingUseContext::Projection);
+                self.visit_place(&place, ctx, location)
```

Comment:
I think this can just be `super_statement`. Not sure where `MutatingUseContext::Projection` came from?

Reply:
that was me trying to guess what would be a proper context for setting a discriminant :P. I'll force push the change.